week_1.py

```diff
@@ -0,0 +1,188 @@
+        self.total_items[item] = discount_item
+        self.show_items()
+
+    def get_total(self): #add all the values in the dictionary to get a total
+        self.total_price = sum(self.total_items.values())
+        print(f'The total of your receipt is £{self.total_price}.')
+
+    def show_items(self): #print all the items in the dictionary
+        for key, values in self.total_items.items():
+            print(key, values)
+
+    def reset_register(self): #reset the register means reset the total_price to 0
+        self.total_items = {}  # None # {'item': 'price'}
+        self.total_price = 0
+        self.discount = 0
+        print(f'The total items are: {self.total_items}\nThe total_price is: {self.total_price}\nThe discount is: {self.discount}')
+
+    def calc_loyalty_points(self): #work out loyalty points
+        loyalty_points = round(self.total_price / 5)
+        return loyalty_points
+
+    def show_loyalty_points(self): #print loyalty points
+        loyalty_points1 = self.calc_loyalty_points()
+        print(f'The number of loyalty points you have are: {loyalty_points1}')
+
+    def calc_loyalty_discount(self): #calc prices using loyalty points
+        discount_price = round(self.total_price - (self.calc_loyalty_points() * 0.2))
+        print(f'The discount price using loyalty points is: £{discount_price}')
+
+# cr = CashRegister()
+# cr.add_item("t-shirt", 5)
+# cr.add_item("shoes", 10)
+# cr.add_item("socks", 15)
+# cr.show_items()
+# print("\n")
+# cr.remove_item("t-shirt")
+# cr.show_items()
+# print("\n")
+# cr.apply_discount(20, "socks")
+# print("\n")
+# cr.get_total()
+# print("\n")
+# cr.show_loyalty_points()
+# print("\n")
+# cr.calc_loyalty_discount()
+# print("\n")
+# cr.reset_register()
+
+
+# EXAMPLE code run:
+
+# ADD
+
+
+"""
+
+TASK 2
+
+Write a base class to represent a student. Below is a starter code. 
+Feel free to add any more new features to your class. 
+As a minimum a student has a name and age and a unique ID.
+
+Create a new subclass from student to represent a concrete student doing a specialization, for example:
+Software Student and Data Science student. 
+
+"""
+
+
+class Student:
+
+    def __init__(self, name, age, id):
+        self.name = name
+        self.age = age
+        self.id = id
+        self.subjects = dict() #{} #subject:grade
+
+class CFGStudent(Student):
+
+    def __init__(self, name, age, id, specialisation, attendance): #specialisation
+        self.name = name
+        self.age = age
+        self.id = id
+        self.specialisation = specialisation
+        self.attendance = attendance
+        Student.__init__(self, self.name, self.age, self.id) #only the attributes from above
+
+    def add_subject(self, subject, grade):
+        self.subjects[subject] = grade
+        return self.subjects
+
+    def remove_subject(self, subject):
+        self.subjects.pop(subject)
+        return self.subjects
+
+    def show_items(self): #show all the subjects and grades
```

Comment:
to be a bit picky, i would be more descriptive with the naming of the function, so maybe show_subjects_and_grades() instead of just show_items().